internal/server/iigointernal/executive.go

```diff
@@ -0,0 +1,148 @@
+package iigointernal
+
+import (
+	"github.com/SOMAS2020/SOMAS2020/internal/common/baseclient"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/gamestate"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/roles"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/voting"
+)
+
+type executive struct {
+	ID               shared.ClientID
+	clientPresident  roles.President
+	budget           shared.Resources
+	speakerSalary    shared.Resources
+	RulesProposals   []string
+	ResourceRequests map[shared.ClientID]shared.Resources
+}
+
+// returnSpeakerSalary returns the salary to the common pool.
+func (e *executive) returnSpeakerSalary() shared.Resources {
+	x := e.speakerSalary
+	e.speakerSalary = 0
+	return x
+}
+
+// Get rule proposals to be voted on from remaining islands
+// Called by orchestration
+func (e *executive) setRuleProposals(rulesProposals []string) {
+	e.RulesProposals = rulesProposals
+}
+
+// Set approved resources request for all the remaining islands
+// Called by orchestration
+func (e *executive) setAllocationRequest(resourceRequests map[shared.ClientID]shared.Resources) {
+	e.ResourceRequests = resourceRequests
+}
+
+// Get rules to be voted on to Speaker
+// Called by orchestration at the end of the turn
+func (e *executive) getRuleForSpeaker() string {
+	e.budget -= serviceCharge
+	result, _ := e.clientPresident.PickRuleToVote(e.RulesProposals)
+	return result
+}
+
+// Send Tax map all the remaining islands
+// Called by orchestration at the end of the turn
+func (e *executive) getTaxMap(islandsResources map[shared.ClientID]shared.Resources) map[shared.ClientID]shared.Resources {
+	e.budget -= serviceCharge
+	result, _ := e.clientPresident.SetTaxationAmount(islandsResources)
+	return result
+}
+
+// broadcastTaxation broadcasts the tax amount decided by the president to all island still in the game.
+func (e *executive) broadcastTaxation(islandsResources map[shared.ClientID]shared.Resources) {
+	e.budget -= serviceCharge
+	taxAmountMap := e.getTaxMap(islandsResources)
+	for _, v := range getIslandAlive() {
+		d := baseclient.Communication{T: baseclient.CommunicationInt, IntegerData: int(taxAmountMap[shared.ClientID(int(v))])}
+		data := make(map[int]baseclient.Communication)
+		data[TaxAmount] = d
```

Comment:
where is TaxAmount defined?

Reply:
We have an enum in utilities.go which has these values defined. I'm going to move it to the base client package so that it's universally accessible and is closer to the communication struct it's designed for